lib/cimp/repo.go

```diff
@@ -0,0 +1,72 @@
+package cimp
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/consul/api"
+)
+
+type Config struct {
+	Address string
+}
+
+type ConsulRepo struct {
+	client *api.Client
+}
+
+func InitRepo(cfg Config) (*ConsulRepo, error) {
+	clientCfg := api.DefaultConfig()
+	clientCfg.Address = cfg.Address
+
+	client, err := api.NewClient(clientCfg)
+	if err != nil {
+		return nil, fmt.Errorf("create consul client: %w", err)
+	}
+
+	return &ConsulRepo{
+		client: client,
+	}, nil
+}
+
+func (repo *ConsulRepo) Save(kv KV) error {
+	for k, v := range kv {
+		kvPair := &api.KVPair{
+			Key:   k,
+			Value: []byte(fmt.Sprintf("%v", v)),
+		}
+		_, err := repo.client.KV().Put(kvPair, nil)
```

Comment:
there is a batch method which would be more effective rather than per key put

Reply:
> there is a batch method which would be more effective rather than per key put

fixed